### ISSUE TITLE
Dogfooding - should classes be Serializable?

### ISSUE THREAD

**Issue by ThrawnCA** [spotbugs/spotbugs#81]
Running FindBugs over the SpotBugs codebase, there are 8 hits for SE_BAD_FIELD, ie a non-serializable field in a Serializable class. Do we really need these classes to be Serializable? If so, we should make it work properly; if not, we shouldn't declare it.

`edu.umd.cs.findbugs.BugInstance`
`edu.umd.cs.findbugs.ba.type.ExceptionSetFactory`
`edu.umd.cs.findbugs.gui2.MainFrameMenu`
`edu.umd.cs.findbugs.gui2.MainFrameTree`
`edu.umd.cs.findbugs.log.Profiler$ClassNameComparator`

**Comment by jsotuyod** [comment]
Most likely than not, none of those needs to be `Serializable`. I think we can remove those `implements` and see what breaks. Either you will get a compile error, or a brand new `SE_BAD_FIELD` report somewhere else on the chain.

The only of those that seems to have any reason to be `Serializable` is `BugInstance` due to the cloud plugins, but that is gone now, so it's most likely useless.

**Comment by mebigfatguy** [comment]
maybe BugInstance was for cloud? no idea

**Comment by jsotuyod** [comment]
I've been looking into this, it seems to stem down from the fact that `MainFrame`, as an awt component, is `Seralizable`.

AWT serialization is not used. It's mainly intended for Applets (not our case) and may be used for a sort-of prevalence, but it's not our intended use case either.

We should probably just mark those fields as `transient` on `MainFrame`, implement `readObject` with something along the lines:

```java
    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
        in.defaultReadObject();
        // re-init non-serializable fields
        logger = new ConsoleLogger(this);
        backgroundExecutor = Executors.newCachedThreadPool();
        mainFrameInitialized = new CountDownLatch(1);
        ....
    }
```

**Comment by mebigfatguy** [comment]
i think you can just rip it all out with impunity

**Comment by jsotuyod** [comment]
Yeah, that could also be, but we would have to suppress the `SE_BAD_FIELD` warnings on the awt compontents.

@ThrawnCA will you take this over now or shall someone else do it?

**Comment by ThrawnCA** [comment]
OK, I've removed the `serialVersionUID` from the class we changed. There are a bunch of other classes that define it but aren't `Serializable`; will get to them later.

**Comment by jsotuyod** [comment]
Resolved on #85
